README.md

```diff
@@ -51,6 +51,15 @@ cd flashinfer/python
 pip install -e .
 ```
 
+when building and testing on A100:
+```bash
+git clone https://github.com/flashinfer-ai/flashinfer.git --recursive
+cd flashinfer/python
+# workaround for undefined symbol `__gmon_start__' on A100
```

Comment:
Can you provide a link to an existing issue (if any)?

Reply:
I encountered this issue when compiling FlashInfer from source on A100 machine with 8 devices (https://www.runpod.io). The image used was `pytorch:2.2.0-py3.10-cuda12.1.1-devel-ubuntu22.04`, and I did not raise a separate issue for it. The method in this PR is a workaround that I verified on runpod.

Reply:
Oh I just noticed your error message, that error appears when binary size is too large, which I should fix these days. Limiting the target CUDA arch is an option to reduce binary size, not only applies to A100. I suppose you might encounter the same issue for other GPU instances.

Reply:
I found similar issues or questions on forums through Google, but their solutions didn't resolve my issue. ref https://www.google.com/search?q=R_X86_64_REX_GOTPCRELX%20against%20undefined%20symbol%20%60__gmon_start__%27

Reply:
> I suppose you might encounter the same issue for other GPU instances.

Yes😂

Reply:
The fundamental solution is to break the CUDAExtension into multiple sudmoules, and compile each of them into a shared object with reasonable size. cc @Yard1 as you might be interested.

Reply:
Currently, the workflow running in the FlashInfer repository is functioning properly. For most users (not FlashInfer developers), using the whl compiled in the workflow is sufficient. If someone is a developer and need to modify code and compile within FlashInfer, like me, before implementing what you mentioned as 'break the CUDAExtension into multiple submodules', perhaps this is an acceptable workaround.

Reply:
So my suggestion here is to keep this note, but mentioning it's for reducing binary size, don't say it's only for A100.
The following function in torch can help user identify their device capability:
https://pytorch.org/docs/stable/generated/torch.cuda.get_device_capability.html#torch.cuda.get_device_capability

Reply:
ok

Reply:
@yzh119 May you help check if the new changes are alright? Thanks.